various-readers/src/main/java/com/dellemc/oe/readers/FlinkSQLReader.java

```diff
@@ -0,0 +1,117 @@
+
+        LOG.info("#######################     SCOPE   ###################### " + scope);
+        LOG.info("#######################     streamName   ###################### " + streamName);
+        LOG.info("#######################     controllerURI   ###################### " + controllerURI);
+        run(scope, streamName, controllerURI);
+    }
+
+    public static void run(String scope, String streamName, URI controllerURI) {
+
+        try {
+            // Read from jason stream
+            streamName = "json-stream";
+            // Create client config
+            PravegaConfig pravegaConfig =  PravegaConfig.fromDefaults()
+                        .withControllerURI(controllerURI)
+                        .withDefaultScope(scope)
+                        .withHostnameValidation(false);
+
+            // create the Pravega input stream (if necessary)
+            Stream stream = Stream.of(scope, streamName);
+
+            // Create Flink Execution environment
+            StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+            env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime);
+            // Read table as stream data
+            StreamTableEnvironment tableEnv = TableEnvironment.getTableEnvironment(env);
+
+            // get the scema
+            Schema   schema = EarthQuakeRecord.getSchema();
+
+            Pravega pravega = new Pravega();
+            pravega.tableSourceReaderBuilder()
+                     .forStream(stream)
+                    .withPravegaConfig(pravegaConfig);
+
+            // Create Table descriptor
+            StreamTableDescriptor desc = tableEnv.connect(pravega)
+                    .withFormat(new Json().failOnMissingField(false).deriveSchema())
+                    .withSchema(schema)
+                    .inAppendMode();
+
+            // Create Table source
+            final Map<String, String> propertiesMap = desc.toProperties();
+            final TableSource<?> source = TableFactoryService.find(StreamTableSourceFactory.class, propertiesMap)
+                    .createStreamTableSource(propertiesMap);
+
+            // Register table resource
+            tableEnv.registerTableSource("earthquakes", source);
+            String sqlQuery = "SELECT DateTime, Latitude, Longitude, Depth, Magnitude, MagType, NbStations, Gap, Distance, RMS, " +
+                    "Source, EventID from earthquakes  where Magnitude > 6"; // Magnitude above 6 treated as high/severe earthquake
+            Table result = tableEnv.sqlQuery(sqlQuery);
+
+            tableEnv.toAppendStream(result, Row.class).printToErr().name("stdout");
+
+            // execute within the Flink environment
+            env.execute("FlinkSQL Reader");
+
+            LOG.info("########## FlinkSQLReader END #############");
+
+        } catch (Exception e) {
+            LOG.error("########## FlinkSQLReader ERROR  #############  "+e.getMessage());
```

Comment:
Change to LOG.error("########## FlinkSQLReader ERROR  #############  ", e);
Remove e.printStackTrace().

Reply:
Removed